src/Aspire.Hosting.NodeJs/NodeExtensions.cs

```diff
@@ -57,6 +57,58 @@ public static IResourceBuilder<NodeAppResource> AddNpmApp(this IDistributedAppli
+    /// <param name="scriptName">The script to execute.</param>
+    /// <param name="args">The arguments to pass to the command.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    private static IResourceBuilder<NodeAppResource> AddGenericNodeApp(this IDistributedApplicationBuilder builder, string name, string workingDirectory, string nodeExecutor, string scriptName, string[]? args = null)
+    {
+        string[] allArgs = args is { Length: > 0 }
+            ? ["run", scriptName, "--", .. args]
+            : ["run", scriptName];
+
+        workingDirectory = PathNormalizer.NormalizePathForCurrentPlatform(Path.Combine(builder.AppHostDirectory, workingDirectory));
+        var resource = new NodeAppResource(name, nodeExecutor, workingDirectory);
+
+        return builder.AddResource(resource)
+                      .WithNodeDefaults()
+                      .WithArgs(allArgs);
+    }
+
+    /// <summary>
+    /// Adds a node application to the application model. Executes the yarn command with the specified script name.
+    /// </summary>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/> to add the resource to.</param>
+    /// <param name="name">The name of the resource.</param>
+    /// <param name="workingDirectory">The working directory to use for the command. If null, the working directory of the current process is used.</param>
+    /// <param name="scriptName">The script to execute.</param>
+    /// <param name="args">The arguments to pass to the command.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<NodeAppResource> AddYarnApp(this IDistributedApplicationBuilder builder, string name, string workingDirectory, string scriptName, string[]? args = null)
+    {
+        return builder.AddGenericNodeApp(name, workingDirectory, "yarn", scriptName, args);
+    }
+
+    /// <summary>
+    /// Adds a node application to the application model. Executes the pnpm command with the specified script name.
+    /// </summary>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/> to add the resource to.</param>
+    /// <param name="name">The name of the resource.</param>
+    /// <param name="workingDirectory">The working directory to use for the command. If null, the working directory of the current process is used.</param>
+    /// <param name="scriptName">The script to execute.</param>
+    /// <param name="args">The arguments to pass to the command.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<NodeAppResource> AddPnpmApp(this IDistributedApplicationBuilder builder, string name, string workingDirectory, string scriptName, string[]? args = null)
+    {
+        return builder.AddGenericNodeApp(name, workingDirectory, "pnpm", scriptName, args);
+    }
+
```

Comment:
todo: I currently have the need for a `AddBunApp`, can you add it? I also think that Deno devs might need a `AddDenoApp`.